src/datagrid/DataGrid.tsx

```diff
@@ -250,6 +290,123 @@ export class DataGrid extends React.PureComponent<DataGridProps, DataGridState>
     };
 
     /* ##########  DataGrid private methods start  ############ */
+
+    // Initialize state of grid
+    private setInitalState() {
+        const {allRows, allColumns} = this.state;
+        let rows = this.updateRowIDs(allRows);
+
+        rows.forEach(function(row) {
+            row["isSelected"] = false;
+            row["isExpanded"] = false;
+        });
+
+        this.setState({
+            allRows: [...rows],
+            allColumns: [...this.updateColumnIDs(allColumns)],
+        });
+    }
+
+    // Initialize state of grid with pagination
+    private setInitalStateForPagination() {
+        let {pagination} = this.state;
+        if (pagination) {
+            const {currentPage, pageSize, totalItems} = pagination;
+            const firstItem = this.getFirstItemIndex(currentPage, pageSize);
+            const lastItem = this.getLastItemIndex(pageSize, totalItems, firstItem);
+
+            pagination.totalPages = this.getTotalPages(totalItems, pageSize);
+            pagination.firstItem = firstItem;
+            pagination.lastItem = lastItem;
+            this.setState({pagination: pagination});
+        }
+    }
+
+    /* ############################# Pagination methods start ####################################*/
+    private getTotalPages = (totalItems: number, pageSize: number) => {
+        return Math.floor((totalItems + pageSize - 1) / pageSize);
+    };
+
+    // Get index of first item in page
+    private getFirstItemIndex = (page: number, pageSize: number) => {
+        return page * pageSize - (pageSize - 1);
+    };
+
+    // Get index of last item in page
+    private getLastItemIndex = (pageSize: number, totalItems: number, firstItem: number) => {
+        return Math.min(firstItem + pageSize - 1, totalItems);
+    };
+
+    // Function to handle change in page sizes
+    private handleSelectPageSize = (evt: React.ChangeEvent<HTMLSelectElement>) => {
+        this.getPage(1, parseInt(evt.target.value));
```

Comment:
Why are we moving back to the first page? What is the scenario when I select size '20' on the 4th page?

Reply:
done